ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/privacy/methods/priv/PrivCreatePrivacyGroup.java

```diff
@@ -55,25 +55,25 @@ public String getName() {
   public JsonRpcResponse response(final JsonRpcRequest request) {
     LOG.trace("Executing {}", RpcMethod.PRIV_CREATE_PRIVACY_GROUP.getMethodName());
 
-    final String[] addresses = parameters.required(request.getParams(), 0, String[].class);
-    final Optional<String> name = parameters.optional(request.getParams(), 1, String.class);
-    final Optional<String> description = parameters.optional(request.getParams(), 2, String.class);
+    final CreatePrivacyGroupParameter parameter =
+        parameters.required(request.getParams(), 0, CreatePrivacyGroupParameter.class);
 
-    LOG.trace("Creating a privacy group with name {} and description {}", name, description);
+    LOG.trace(
+        "Creating a privacy group with name {} and description {}",
+        parameter.getName(),
+        parameter.getDescription());
 
     final CreatePrivacyGroupRequest createPrivacyGroupRequest =
         new CreatePrivacyGroupRequest(
-            addresses,
+            parameter.getAddresses(),
             privacyParameters.getEnclavePublicKey(),
-            name.orElse(null),
-            description.orElse(null));
+            parameter.getName(),
+            parameter.getDescription());
     final PrivacyGroup response;
     try {
       response = enclave.createPrivacyGroup(createPrivacyGroupRequest);
     } catch (Exception e) {
-      LOG.error("Failed to fetch transaction from Enclave with error " + e.getMessage());
-      LOG.error(e);
-      return new JsonRpcSuccessResponse(request.getId(), JsonRpcError.CREATE_PRIVACY_GROUP_ERROR);
+      return new JsonRpcErrorResponse(request.getId(), JsonRpcError.CREATE_PRIVACY_GROUP_ERROR);
```

Comment:
Not sure if I suggested removing this log - but its probably worth keeping the log of the error ... in another patchset I deliberately removed low-level logging on the assumption it would be caught in the Json Rpc Handler, and logged there (thus there may not be good logging below this point).